src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -204,15 +206,25 @@ private static bool TryParseStatusLine(HttpResponseMessage response, string resp
                 return false;
             }
 
+	    //Clear the header if status line is recieved again. This signifies that there are multiple response headers (like in redirection).
+
+            response.Headers.Clear();
+            
+            response.Content.Headers.Clear();        
```

Comment:
Is it possible the response object has been handed out to other code at this point?  I'm wondering if this Clear operation is safe.